.github/scripts/create-vm.py

```diff
@@ -0,0 +1,462 @@
+    def __call__(self, parser, namespace, values, option_string=None):  # noqa: U100
+        kv_dict = {}
+        for item in values.split(","):
+            key, value = item.split("=")
+            kv_dict[key] = value
+        setattr(namespace, self.dest, kv_dict)
+
+
+def fetch_github_team_public_keys(gh, github_org, team_slug):
+    org = gh.get_organization(github_org)
+    team = org.get_team_by_slug(team_slug)
+    members = [member for member in team.get_members()]
+
+    ssh_keys = []
+    logging.info(
+        f"Fetching SSH keys for members: {[member.login for member in members]}"
+    )
+    for member in members:
+        member_keys = []
+        for key in member.get_keys():
+            member_keys.append(key.key)
+            ssh_keys.append(key.key)
+
+        logging.debug(f"Fetched {len(member_keys)} SSH keys for {member.login}")
+
+    logging.debug(f"Fetched SSH keys: {ssh_keys}")
+    return ssh_keys
+
+
+def generate_cloud_init_script(user, ssh_keys, owner, repo, token, version, label):
+
+    script = [
+        "#!/bin/bash",
+        "set -x",
+        "mkdir -p /actions-runner && cd /actions-runner",
+        'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64";; esac',
+        f"export FILENAME=actions-runner-linux-${{ARCH}}-{version}.tar.gz",
+        # https://github.com/actions/runner/releases/download/v2.314.1/actions-runner-linux-x64-2.314.1.tar.gz
+        f'[ -f "$FILENAME" ] || curl -O -L "https://github.com/actions/runner/releases/download/v{version}/$FILENAME"',
+        'tar xzf "./$FILENAME"',
+        "export RUNNER_ALLOW_RUNASROOT=1",
+        f"./config.sh  --labels {label} --url https://github.com/{owner}/{repo} --token {token}",
+        "./run.sh",
+    ]
+
+    cloud_init = {}
+    cloud_init["ssh_pwauth"] = False
+    cloud_init["users"] = [
+        {
+            "name": user,
+            "sudo": "ALL=(ALL) NOPASSWD:ALL",
+            "passwd": os.environ["VM_USER_PASSWD"],
+            "shell": "/bin/bash",
+            "ssh_authorized_keys": ssh_keys,
+        }
+    ]
+    cloud_init["runcmd"] = ["\n".join(script)]
+    logging.info(
+        f"Cloud-init: \n{yaml.dump(cloud_init, default_flow_style=False, width=math.inf)}".replace(
+            token, "****"
+        )
+    )
+    return (
+        "#cloud-config\n"
+        + yaml.dump(cloud_init, default_flow_style=False, width=math.inf),  # noqa: W503
+        ssh_keys,
+    )
+
+
+def get_runner_token(github_repo_owner, github_repo, github_token):
+    result = requests.post(
+        f"https://api.github.com/repos/{github_repo_owner}/{github_repo}/actions/runners/registration-token",
+        headers={
+            "Authorization": f"Bearer {github_token}",
+            "Accept": "application/vnd.github+json",
+            "X-Github-Api-Version": "2022-11-28",
+        },
+    ).json()
+
+    token = result.get("token")
+    if token:
+        # Mask the token in the logs
+        print(f"::add-mask::{token}")
```

Comment:
Avoid print's if you use logging

Reply:
This is not for debugging, it is control string to tell GA that this string is masked.